contrib/swank-quicklisp.lisp

```diff
@@ -6,12 +6,16 @@
 
 (in-package :swank)
 
-(defslimefun list-quicklisp-systems ()
+(defslimefun list-quicklisp-systems (with-local-systems)
   "Returns the Quicklisp systems list."
   (if (member :quicklisp *features*)
       (let ((ql-dist-name (find-symbol "NAME" "QL-DIST"))
-            (ql-system-list (find-symbol "SYSTEM-LIST" "QL")))
-        (mapcar ql-dist-name (funcall ql-system-list)))
+            (ql-system-list (find-symbol "SYSTEM-LIST" "QL"))
+            (ql-local-system-list
+             (find-symbol "LIST-LOCAL-SYSTEMS" "QL")))
+        (append (when with-local-systems
+                  (funcall ql-local-system-list))
+                (mapcar ql-dist-name (funcall ql-system-list))))
```

Comment:
Should we sort them by name or something?

Reply:
I don't think it is necessary: completing-read seems to sort the items (in this case the appended list in the above code block) automatically.